### ISSUE TITLE
Accumulated Local Effects (ALE)

### ISSUE THREAD

**Issue by jklaise** [SeldonIO/alibi#151]
A global explanation method, see https://christophm.github.io/interpretable-ml-book/ale.html . Now that we have distance measures between categories in Alibi we are in a position to implement this for categorical features as well as numerical ones.

**Comment by jklaise** [comment]
Minimal requirements:
- 1-way numerical ALE
- 2-way numerical ALE
- 1-way categorical ALE
- Clean visualisations for all of the above

Secondary requirements (probably another PR):
- 2-way categorical ALE
- 2-way numerical/categorical mixed ALE
- caching of previously calculated results
- batching for large datasets
- ability to do all 1-ways/2-ways in parallel

**Comment by jklaise** [comment]
Initial implementation in #152 .
